src/core/ucc_constructor.c

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+#include "ucc_global_opts.h"
+#include "utils/ucc_malloc.h"
+#include <link.h>
+#include <dlfcn.h>
+#include <string.h>
+
+static int callback(struct dl_phdr_info *info, size_t size, void *data)
+{
+    char *str;
+    char *component_path;
+    if (NULL != (str = strstr(info->dlpi_name, "libucc.so"))) {
+        int pos        = (int)(str - info->dlpi_name);
+        component_path = (char *)ucc_malloc(pos + 8);
+        strncpy(component_path, info->dlpi_name, pos);
+        component_path[pos] = '\0';
+        strcat(component_path, "ucc");
+        ucc_global_config.component_path =
+            ucc_global_config.component_path_default = component_path;
+    }
+    return 0;
+}
+
+static void get_default_lib_path()
+{
+    dl_iterate_phdr(callback, NULL);
+}
+
+ucc_status_t ucc_constructor(void)
+{
+    ucc_status_t status;
+    if (!ucc_global_config.initialized) {
+        ucc_global_config.initialized            = 1;
+        ucc_global_config.component_path_default = NULL;
+
+        status = ucc_config_parser_fill_opts(
+            &ucc_global_config, ucc_global_config_table, "UCC_", NULL, 1);
```

Comment:
Minor: ident (I'm not sure, if this another clang-format artifact)

Reply:
@manjugv what about indent here? you mean that fn arguments are on the next line after open brace? - this is how our clang format is configured.

Reply:
I thought so. What do you think is better here? follow clang-format or align it with the braces. I'm always used to aligning with braces. For me, I want us to be consistent. I don't really one way or another.